kale/prepdata/supergraph_construct.py

```diff
@@ -0,0 +1,250 @@
+            sort_index = torch.argsort(self.edge_type)
+            self.edge_index = self.edge_index[:, sort_index]
+            self.edge_type = self.edge_type[sort_index]
+
+            self.__get_range_list__()
+
+    def __get_range_list__(self):
+        """get the range of edge types"""
+        idx = 0
+        range_list = [[0, 0]]
+
+        for i, et in enumerate(self.edge_type):
+            if et != idx:
+                idx = et
+                range_list[-1][1] = i
+                range_list.append([i, i])
+
+        range_list[-1][1] = i + 1
+
+        self.range_list = torch.tensor(range_list)
+
+    def __repr__(self) -> str:
+        return f"SuperVertex(\n    name={self.name}, \n    node_feat={self.node_feat.shape}, \n    edge_index={self.edge_index.shape}, \n    num_edge_type={self.num_edge_type})"
+
+    def add_in_supervertex(self, vertex_name: str):
+        self.in_supervertex_list.append(vertex_name)
+
+    def add_out_supervertex(self, vertex_name: str):
+        self.out_supervertex_list.append(vertex_name)
+
+
+class SuperEdge(object):
+    r"""
+    The superedge structure in GripNet. Each superedge is a bipartite subgraph containing nodes from two categories
+    forming two node sets, connected by edges between them. The superedge can be regards as a heterogeneous graph
+    connecting two supervertices.
+
+    Args:
+        source_supervertex (str): the name of the source supervertex.
+        target_supervertex (str): the name of the target supervertex.
+        edge_index (torch.Tensor): edge indices in COO format with shape [2, #edges]. The first row is the index of
+            source nodes, and the second row is the index of target nodes.
+        edge_weight (torch.Tensor, optional): one-dimensional weight for each edge. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        source_supervertex: str,
+        target_supervertex: str,
+        edge_index: torch.Tensor,
+        edge_weight: torch.Tensor = None,
+    ) -> None:
+
+        self.direction = (source_supervertex, target_supervertex)
+        self.source_supervertex = source_supervertex
+        self.target_supervertex = target_supervertex
+        self.edge_index = edge_index
+        self.edge_weight = edge_weight
+
+    def __repr__(self) -> str:
+        return f"SuperEdges(\n    edge_direction={self.source_supervertex}->{self.target_supervertex}, \n    edge_index={self.edge_index.shape})"
+
+
+class SuperVertexParaSetting(object):
+    r"""Parameter settings for each supervertex.
+
+        Args:
+            supervertex_name (str): the name of the supervertex.
+            inter_feat_dim (int): the dimension of the output of the internal feature layer.
+            inter_agg_dim (List[int]): the output dimensions of a sequence of internal aggregation layers.
+            exter_agg_dim (Dict[str, int], optional): the dimension of received message vector from parient supervertices.
```

Comment:
parient --> parent?

Reply:
Fixed in [5c193a3](https://github.com/pykale/pykale/pull/338/commits/5c193a39c6e3d76aa9249ae5ff9546a17d6c60a2).